src/components/GoalForm/GoalForm.tsx

```diff
@@ -134,13 +134,37 @@ export const GoalForm: React.FC<GoalFormProps> = ({
             estimate: null,
             tags: [],
             id: null,
+            mode: goalType.current,
         },
     });
 
     const parentWatcher = watch('parent');
+    const mode = watch('mode');
     const tagsWatcher: TagModel[] = watch('tags');
     const errorsResolver = errorsProvider(errors, isSubmitted);
 
+    useEffect(() => {
+        goalType.current = personal ? goalTypeMap.personal : goalTypeMap.default;
+    }, [personal]);
+
+    useEffect(() => {
+        const parentSubWatch = watch(({ mode }, { name }) => {
+            if (name === 'mode') {
+                if (mode === goalTypeMap.personal) {
+                    setValue('parent', null);
+                } else {
+                    setValue('parent', parent);
+                }
+                resetField('state');
+                resetField('parent');
+            }
+        });
+
+        return () => {
+            parentSubWatch.unsubscribe();
+        };
+    }, [watch, setValue, parent, resetField]);
```

Comment:
Why can't we use just `useEffect` with `mode` in deps to reset fields?
I think it can be easier..

Reply:
Do you propose to use decision like

```ts
const { watch, setValue, resetField } = useForm(/* */)

const mode = watch('mode');

useEffect(() => {
    if (mode === 'personal') {
        /* * /
    } else {
        /* */
    }
}, [mode, setValue, resetField])
```
?

I don't think it's gotten any better